packages/react-components/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -26,15 +26,13 @@ const useRootStyles = makeStyles({
 
 const useInputStyles = makeStyles({
   base: {
-    position: 'absolute',
-    left: 0,
-    top: 0,
-    width: '100%',
-    height: '100%',
     boxSizing: 'border-box',
+    cursor: 'pointer',
+    height: '100%',
     ...shorthands.margin(0),
     opacity: 0,
-    cursor: 'pointer',
+    position: 'absolute',
+    top: 0,
 
     // When unchecked, hide the the checkmark icon (child of the indicator slot)
     [`:not(:checked):not(:indeterminate) ~ .${checkboxClassNames.indicator} > *`]: {
```

Comment:
Followed up offline, but to recap: I don't think the hover/focus/etc. styles below will work anymore when hovering over the label, after this change. I can't think of a simple solution; the best I can think of is adding some additional CSS variables like "hoverLabelColor", which gets set by the input based on its enabled/checked/indeterminate state. Then the label would have a hover selector that sets that color.

(A little background on why the styles are factored like this: It's not necessarily possible to track the checked state for the radio button in javascript, which is one of the reasons this uses css selectors for those states instead of tracking the state in code. The disabled state is also possibly inherited from a parent group/form, so tracking that in code is also not simple.)

Reply:
I double-checked and this is apparently still working. My guess is that the browser treats hovering/pressing over the label that's attached to an input element the same way as it treats hovering/pressing over the input element itself.